cumulus/parachains/runtimes/assets/common/src/erc20_transactor.rs

```diff
@@ -0,0 +1,196 @@
+			IERC20::transferCall { to: checking_address, value: EU256::from(amount) }.abi_encode();
+		let ContractResult { result, gas_consumed, .. } = pallet_revive::Pallet::<T>::bare_call(
+			T::RuntimeOrigin::signed(who.clone()),
+			asset_id,
+			BalanceOf::<T>::zero(),
+			gas_limit,
+			DepositLimit::Unchecked,
+			data,
+		);
+		// We need to return this surplus for the executor to allow refunding it.
+		let surplus = gas_limit.saturating_sub(gas_consumed);
+		tracing::trace!(target: "xcm::transactor::erc20::withdraw", ?gas_consumed, ?surplus, "Gas consumed");
+		if let Ok(return_value) = result {
+			tracing::trace!(target: "xcm::transactor::erc20::withdraw", ?return_value, "Return value by withdraw_asset");
+			if return_value.did_revert() {
+				tracing::debug!(target: "xcm::transactor::erc20::withdraw", "ERC20 contract reverted");
+				Err(XcmError::FailedToTransactAsset("ERC20 contract reverted"))
+			} else {
+				let is_success = bool::abi_decode(&return_value.data, false).map_err(|error| {
+					tracing::debug!(target: "xcm::transactor::erc20::withdraw", ?error, "ERC20 contract result couldn't decode");
+					XcmError::FailedToTransactAsset("ERC20 contract result couldn't decode")
+				})?;
+				if is_success {
+					Ok((what.clone().into(), surplus))
+				} else {
+					tracing::debug!(target: "xcm::transactor::erc20::withdraw", "contract transfer failed");
+					Err(XcmError::FailedToTransactAsset("ERC20 contract transfer failed"))
+				}
+			}
+		} else {
+			tracing::debug!(target: "xcm::transactor::erc20::withdraw", ?result, "Error");
+			// This error could've been duplicate smart contract, out of gas, etc.
+			// If the issue is gas, there's nothing the user can change in the XCM
+			// that will make this work since there's a hardcoded gas limit.
+			Err(XcmError::FailedToTransactAsset("ERC20 contract execution errored"))
+		}
+	}
+
+	fn deposit_asset_with_surplus(
+		what: &Asset,
+		who: &Location,
+		_context: Option<&XcmContext>,
+	) -> Result<Weight, XcmError> {
+		tracing::trace!(
+			target: "xcm::transactor::erc20::deposit",
+			?what, ?who,
+		);
+		let (asset_id, amount) = Matcher::matches_fungibles(what)?;
+		let who = AccountIdConverter::convert_location(who)
+			.ok_or(MatchError::AccountIdConversionFailed)?;
+		// We need to map the 32 byte beneficiary account to a 20 byte account.
+		let eth_address = T::AddressMapper::to_address(&who);
+		let address = Address::from(Into::<[u8; 20]>::into(eth_address));
+		// To deposit, we actually transfer from the checking account to the beneficiary.
+		// We do this using the solidity ERC20 interface.
+		let data = IERC20::transferCall { to: address, value: EU256::from(amount) }.abi_encode();
+		let gas_limit = GasLimit::get();
+		let ContractResult { result, gas_consumed, .. } = pallet_revive::Pallet::<T>::bare_call(
+			T::RuntimeOrigin::signed(TransfersCheckingAccount::get()),
+			asset_id,
+			BalanceOf::<T>::zero(),
+			gas_limit,
+			DepositLimit::Unchecked,
```

Comment:
Should never be passed for on-chain code. This is only meant for dry-runs. We need to rename this variant as it is unsafe to use for on-chain code. It doesn't mean "allow all". It actually doesn't collect the storage deposit.

You can use:
```
DepositLimit::Balance(Balance::MAX)
```